### PR TITLE
Fix padding formula

### PR DESCRIPTION
**Pull request by nithinraok**

### CODE REVIEW

File: nemo/collections/asr/parts/jasper.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def compute_new_kernel_size(kernel_size, kernel_width):
 def get_same_padding(kernel_size, stride, dilation) -> int:
     if stride > 1 and dilation > 1:
         raise ValueError("Only stride OR dilation may be greater than 1")
-    return (dilation * (kernel_size - 1)) // 2
+    return (kernel_size + (kernel_size - 1) * (dilation - 1) - 1) // 2
```
Comment (mousebaiker) [inline nemo/collections/asr/parts/jasper.py:74]: This formula is the exact same formula as the old one. Just expand:
`k + (k - 1) * (d - 1) - 1 = k + kd - k - d  + 1 - 1 = kd - d = d * (k - 1)`

You can also verify numerically:
```
def p1(k, d):
  return (d * (k - 1))//2

def p2(k ,d):
   return (k + (k - 1) * (d - 1) - 1) // 2


for k in range(1, 1001):
   for d in range(1, 1000):
     if p1(k, d) != p2(k, d):
       print(k, d)
```

I would suggest keeping the old formula and just adding tests in this PR.

File: tests/collections/asr/test_jasper_block.py [hunk 2]
```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import random
```
Comment (mousebaiker) [inline tests/collections/asr/test_jasper_block.py:15]: This is probably not needed since you're using `torch.rand`.

**Review by mousebaiker — CHANGES_REQUESTED** [review]

File: nemo/collections/asr/parts/jasper.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/asr/parts/jasper.py:74]: @nithinraok The new formula here and the one before are identical. The expanded formula is far harder to interpret and extend to the asymmetric case as well. Revert and just add the tests. As a sanity check, also add the above code snippet as a test.

Comment (nithinraok) [reply nemo/collections/asr/parts/jasper.py:74]: Sorry was basing my PR on my local changes from rc1 branch which has a previous formula of `(d*k)//2 -1` didn't notice it was fixed in main. changed the formula to its shorter form of `d*(k-1)` funny that I missed it. Will update with tests only

File: tests/collections/asr/test_jasper_block.py (hunk 2 above)
Comment (nithinraok) [reply tests/collections/asr/test_jasper_block.py:15]: I thought of using random.randrange to generate random kernel and dilation sizes, but later found out we cannot always find a padding size for conv1d to make input output same for some even kernel sizes like 4 and dilation 3. 
There must be a reason why pytorch didn't support this to append zeros?

Comment (titu1994) [reply tests/collections/asr/test_jasper_block.py:15]: We force all kernels to odd size so don't consider even kernel shapes.

File: tests/collections/asr/test_jasper_block.py [hunk 3]
```diff
@@ -273,3 +273,23 @@ def test_residual_block_asymmetric_pad_future_context_fallback(self):
         assert ylen[0] == 131
         assert block.mconv[0].pad_layer is None
         assert block.mconv[0]._padding == config['kernel_size'][0] // 2
+
+    @pytest.mark.unit
+    def test_padding_size_conv1d(self):
+        input_size = 1024
```
Comment (titu1994) [inline tests/collections/asr/test_jasper_block.py:279]: For a test do we really need such large arrays?

File: tests/collections/asr/test_jasper_block.py [hunk 4]
```diff
@@ -273,3 +273,23 @@ def test_residual_block_asymmetric_pad_future_context_fallback(self):
...
+        output_size = 1024
```
Comment (titu1994) [inline tests/collections/asr/test_jasper_block.py:280]: Same as ^

**Review by titu1994 — APPROVED** [review]

**Review by mousebaiker — APPROVED** [review]
LGTM.

**Pull request merged**